### ISSUE TITLE
intersectsSphere() - Raycasting

### ISSUE THREAD

**Issue by vedhant** [stalgiag/p5.xr#43]
```
intersectsSphere(Number radius, [screenX], [screenY])
intersectsSphere(Number radius, rayObject ray) 
- returns Boolean
```

**Description**
Checks whether a ray intersects a sphere. The radius of the sphere is passed in the 1st argument and the sphere will be applied the same transformations as present while calling the function.
The 2nd argument is optional. If it is not present, ray will be emerged from mouse position and so intersection with sphere can be tested with mouse. Ray can be emerged from viewer position in the direction specified by parameters screenX and screenY. ScreenX and screenY should be specified in device normalised form ` -1 <= screenX, screenY <=1`. ScreenX and screenY defaults to 0. If a user wishes to use a custom ray, he/she can call generateRay() and pass it in the 2nd paramenter.

**Example**
```javascript
let randomx=[], randomy=[];

function preload() {
  createVRCanvas();
}

function setup() {
  setVRBackgroundColor(200, 0, 150);
  for(let i=0; i<5; ++i) {
    randomx[i] = random(-500, 500);
    randomy[i] = random(-500, 500);
  }
}

function draw() {
  setViewerPosition(0, 0, 400);
  for(let i=0; i<5; ++i) {
    push();
    translate(randomx[i], randomy[i]);
    fill('red');
    if(intersectsSphere(70)) {
      fill('blue');
    }
    sphere(70);
    pop();
  }
}

```
**High-level implementation overview**

1. Retrieve ray coordinates in view space. If it is from mouse, inverse projection matrix will be used. For custom rays, the ray can be used directly.
2. Apply current transformation to sphere collider.
3. Run ray-sphere intersection algorithm.

**Comment by vedhant** [comment]
There is one flaw in the api spec. I have considered mouse position if no ray is passed but there is no mouse in a xr scene. So, I will be taking screen coordinates from the user instead.

**Comment by stalgiag** [comment]
Typically in VR situations, the mouse position is equivalent to the center point between the eyes. The eye offset converges in the distance at a ray that is in the center. To my knowledge, this is how most gaze based interaction works.

**Comment by vedhant** [comment]
@stalgiag you are right. I have fixed it in #48 .The manual example in #48 for intersectsSphere() is very similar to gaze based selection.

**Comment by stalgiag** [comment]
closed via #48
